### PR TITLE
Allow crm_node -l/-p to be used from Pacemaker Remote nodes

### PR DESCRIPTION
**Pull request by kgaillot**
This builds on the recently introduced universal daemon IPC model to move crm_node's -l/-p functionality from pacemakerd to pacemaker-controld, which allows it to work on Pacemaker Remote nodes (which proxy controller connections but not pacemakerd connections). It also contains some ancillary improvements in pacemakerd.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this?

@wenningerk , the pacemakerd changes here will likely conflict with yours, but hopefully it will be straightforward to merge them

File: daemons/controld/controld_messages.c [hunk 1]
```diff
@@ -375,10 +375,11 @@ relay_message(xmlNode * msg, gboolean originated_locally)
             is_local = 0;
 
         } else if (is_for_crm) {
-            if (safe_str_eq(task, CRM_OP_NODE_INFO)) {
+            if (pcmk__str_any_of(task, CRM_OP_NODE_INFO,
+                                 PCMK__CONTROLD_CMD_NODES, NULL)) {
```
Comment (clumens) [inline daemons/controld/controld_messages.c:379]: I'm glad this is already being useful.

File: tools/crm_node.c [hunk 2]
```diff
@@ -130,6 +130,16 @@ remove_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError *
...
+static gint
+sort_node(gconstpointer a, gconstpointer b)
+{
+    const pcmk_controld_api_node_t *node_a = a;
+    const pcmk_controld_api_node_t *node_b = b;
+
+    return strcmp((node_a->uname? node_a->uname : ""),
+                  (node_b->uname? node_b->uname : ""));
+}
+
```
Comment (clumens) [inline tools/crm_node.c:142]: libpe_status has the exhaustively long sort_node_uname.  However, crm_node doesn't currently link against that so I guess we don't want to add the depenency.  On the other hand, that function does seem like something that is more broadly useful.

File: daemons/pacemakerd/pacemakerd.c [hunk 3]
```diff
@@ -1099,25 +1124,7 @@ main(int argc, char **argv)
...
-
-        rc = setrlimit(RLIMIT_CORE, &cores);
-        if (rc < 0) {
-            crm_perror(LOG_ERR,
-                       "Core file generation will remain disabled."
-                       " Core files are an important diagnostic tool, so"
-                       " please consider enabling them by default.");
-        }
-    }
+    remove_core_file_limit();
```
Comment (clumens) [inline daemons/pacemakerd/pacemakerd.c:1107]: This could go into its own commit.

File: daemons/pacemakerd/pcmkd_corosync.c [hunk 4]
```diff
@@ -85,7 +83,7 @@ cluster_disconnect_cfg(void)
 	code;						\
 	if(rc == CS_ERR_TRY_AGAIN || rc == CS_ERR_QUEUE_FULL) {  \
 	    counter++;					\
-	    crm_debug("Retrying operation after %ds", counter);	\
+	    crm_debug("Retrying corosync operation after %ds", counter);    \
```
Comment (clumens) [inline daemons/pacemakerd/pcmkd_corosync.c:86]: In the rest of the log messages I've seen in this commit, "corosync" is capitalized.

File: tools/crm_node.c (hunk 2 above)
Comment (kgaillot) [reply tools/crm_node.c:142]: Ah I had forgot about that one ... that is much better. It operates on a different structure, however I can functionize the actual comparison part and put that in libcrmcommon.

File: daemons/pacemakerd/pacemakerd.c (hunk 3 above)
Comment (kgaillot) [reply daemons/pacemakerd/pacemakerd.c:1107]: good point

File: daemons/pacemakerd/pcmkd_corosync.c (hunk 4 above)
Comment (kgaillot) [reply daemons/pacemakerd/pcmkd_corosync.c:86]: Yeah, we've been really inconsistent about that (for "pacemaker" too). I've even seen "CoroSync" in older documentation. It looks like the code base leans a little more to the init-cap side, so I'll go with that ...

File: tools/crm_node.c (hunk 2 above)
Comment (kgaillot) [reply tools/crm_node.c:142]: FYI the reason sort_node_uname() is long is so that we sort nodes like "node8 node9 node10" rather than "node10 node8 node9".

**Comment by kgaillot** [comment]
updated and re-tested

**Comment by kgaillot** [comment]
retest this please
